src/rpc/misc.cpp

```diff
@@ -45,10 +45,15 @@ using namespace std;
  **/
 UniValue getinfo(const UniValue& params, bool fHelp)
 {
-    if (fHelp || params.size() != 0)
+    if (fHelp || params.size() > 1)
         throw runtime_error(
-            "getinfo\n"
+            "getinfo ( asOfHeight )\n"
             "Returns an object containing various state info.\n"
+            "\nArguments:\n"
+            "1. asOfHeight       (numeric, optional) Execute the query as if it were run when\n"
+            "                    the blockchain was at the height specified by this argument.\n"
+            "                    The default is to use the entire blockchain that the node is\n"
+            "                    aware of.\n"
```

Comment:
Document what happens if `asOfHeight` is greater than the height of the last known block.

Reply:
So, what _does_ happen is it just uses the current block height (that is checked down in `GetDepthInMainChainINTERNAL`). The other option is to error, but that seems unnecessary.

Related to this, is the question of a default value. If we maintain `asOfHeight` as the last parameter, it can always be omitted, which is what I was thinking at the time. But, if we end up with parameters after it, then using later parameters would require _some_ default to be used. Picking a high number isn’t going to be reliable, so perhaps a negative number could be used? Or perhaps the question can be deferred until we cross that bridge.

Reply:
What I’ve changed locally so far here is:
• treat “future” height as current height (what it was already doing)
• treat negative height as unset

Reply:
Now treat negative height (and zero height) as an error, with `"*"` as default.